main.js

```diff
@@ -0,0 +1,42 @@
+var getTreeHeight = document.getElementById("enter-height");
+var getTreeChar = document.getElementById("enter-char");
+var button = document.getElementById("tree-button");
+var growTree = "";
+
+function checkUserInput (clickEvent) {
+	if (getTreeHeight.value === "" && getTreeChar.value === "") {
+		alert("You must fill out both fields!");
+	} else {
+		tree();
+	}
+}
+
+function tree () {
+	var treeElements = {
+		height: getTreeHeight.value,
+		character: getTreeChar.value
+	};
+	treeMath(treeElements);
+}
+
+function treeMath (makeTree) {
+	for (var t=0; t<makeTree.height; t++) {
```

Comment:
I'm thinking _this_ right here is why you are having the crashing issues. When you have nested for-loops, you have to define different variables (eg, they can't all be t) because then it is redefining "t" each time it enters one of the nested forloops, which could be pretty much blowing up your javascript.